src/box/lua/config/applier/box_cfg.lua

```diff
@@ -475,7 +444,141 @@ local function apply(config)
+        box.on_commit(function()
+            local watch_key = 'box_apply_names'
+            local names = box.watch_once(watch_key) or {}
+            table.insert(names, new[2])
+            box.broadcast(watch_key, names)
+        end)
+    end
+end
+
+local function names_cluster_on_replace(old, new)
+    if old[3] == nil and new[3] ~= nil then
+        box.on_commit(function()
+            local watch_key = 'box_apply_names'
+            local names = box.watch_once(watch_key) or {}
+            table.insert(names, new[3])
+            box.broadcast(watch_key, names)
+        end)
+    end
+end
+
+local function names_box_cfg(cfg)
+    local ok, err = pcall(function()
+        box.cfg(cfg)
+    end)
+
+    -- We don't consider failed box.cfg a critical error,
+    -- delete a warning anyway. box.info shows correct info.
+    if not ok then
+        log.warn('Failed to apply name in box.cfg: %s', err)
+    end
+end
+
+local function post_apply(config)
+    local configdata = config._configdata
+    if table.equals(configdata._missing_names, {}) then
+        return
+    end
+
+    local names_apply_fiber = fiber.new(names_apply_f, configdata)
+
+    -- Set triggers to notify apply_name watcher
+    box.space._cluster:on_replace(names_cluster_on_replace)
+    box.space._schema:on_replace(names_schema_on_replace)
+
+    local upgrade_on_replace = nil
+    local version = mkversion.get()
+    if version < mkversion(3, 0, 0) then
+        upgrade_on_replace = box.space._schema:on_replace(function(old, new)
+            if old == nil or new == nil then
+                return
+            end
+
+            local version_3 = mkversion(3, 0, 0)
+            local old_version = mkversion.from_tuple(old)
+            local new_version = mkversion.from_tuple(new)
+            if old_version < version_3 and new_version >= version_3 then
+                box.on_commit(function()
+                    pcall(names_apply_fiber.wakeup, names_apply_fiber)
+                end)
+            end
+        end)
+    end
+
+    local watch
+    local config_names = configdata:names()
+    watch = box.watch('box_apply_names', function(_, names)
+        -- The function is executed, when created, even if no broadcasts
+        -- were done. Do nothing in such case.
+        if names == nil then
+            return
+        end
+
+        for _, name in ipairs(names) do
+            if not configdata._missing_names[name] then
+                goto continue
+            end
+            -- Initialize box.cfg.replicaset/instance_name
+            if name == config_names.replicaset_name then
+                names_box_cfg({replicaset_name = name})
+            end
+            if name == config_names.instance_name then
+                names_box_cfg({instance_name = name})
+            end
+
+            -- Delete name from _missing_names and drop the associated alert.
+            configdata._missing_names[name] = nil
+            config:_alert_drop(name)
```

Comment:
As proposed by @Totktonada, alert must be dropped,when replica is deleted from replicaset (`_cluster`)

Reply:
Done